Snabble/Core/Metadata/Project.swift

```diff
@@ -126,8 +136,17 @@ public struct QRCodeConfig: Decodable {
     static let qrCodeMax = 2953
 
     var effectiveMaxCodes: Int {
-        let leaveRoom = self.nextCode != nil || self.nextCodeWithCheck != nil || self.finalCode != nil || self.manualDiscountFinalCode != nil
-        return self.maxCodes - (leaveRoom ? 1 : 0)
+        switch (finalCode, manualDiscountFinalCode) {
+        case (.some, .some): return maxCodes - 2
+        case (.some, .none), (.none, .some): return maxCodes - 1
+        case (.none, .none): ()
+        }
+
+        if nextCode != nil || self.nextCodeWithCheck != nil {
+            return maxCodes - 1
+        }
```

Comment:
Wenn `nextCode` und `nextCodeWithCheck` beides nicht `nil` ist müsste es dann nicht um zwei reduziert werden?

Reply:
Nein, die schliessen sich gegenseitig aus.

Reply:
Bedeutet vom Server kann nur einer kommen?

Reply:
Hintergrund der Frage ist, sollen wir es im Parsing definieren und einen höher Gewichten, damit es keine Probleme in der App gibt falls doch mal zwei kommen?

Reply:
Nein da können schon beide da sein, die Existenz von `nextCodeWithCheck` triggert auch ein besonderes Verhalten in der Generierung. Die ersten N-1 Blocks in einer Serie von Blocks haben aber immer nur genau einen "next code", entweder den `nextCode` oder den `nextCodeWithCheck`.